### DISCUSSION TITLE
Joints in custom parts

### DISCUSSION THREAD

**Discussion by mortisj** [gumyr/build123d#822] [General]
Based on a recent discussion about creating custom part objects I tried to convert some existing naive part classes to subclass BasePartObject:
```
class CustomCylinder(bd.BasePartObject):
    def __init__(self):
        part = bd.Cylinder(radius = 2, height = 2)
        bd.RigidJoint(
            label = "bottom",
            to_part = part,
            joint_location = bd.Location((0, 0, -1), (0, 0, 0)))
        super().__init__(part=part)
```
After I did this any attempt to connect_to an instance of the custom part silently failed.  The call succeeded but the part was not moved.

The trick was to change to order of things slightly:
```
class CustomCylinder(bd.BasePartObject):
    def __init__(self):
        part = bd.Cylinder(radius = 2, height = 2)
        super().__init__(part=part)
        bd.RigidJoint(
            label = "bottom",
            to_part = self,
            joint_location = bd.Location((0, 0, -1), (0, 0, 0)))
```
With the first (incorrect) version it is strange that the connect_to succeeded without a KeyError.  It looks like the super.__init__() copies the joints dict from the subpart to the new object, but it doesn't adjust the joints' parent attribute appropriately.  I'm not sure if that is a bug, but it certainly tripped me up!

Cheers.

**Comment by gumyr** [comment]
I made an Issue from this discussion - it's a bug.  Thanks for pointing it out.
